packages/playwright-core/src/server/frames.ts

```diff
@@ -965,20 +972,25 @@ export class Frame extends SdkObject {
   async retryWithProgress<R>(
     progress: Progress,
     selector: string,
-    options: types.StrictOptions & types.TimeoutOptions,
-    action: (frame: Frame, info: SelectorInfo, continuePolling: symbol) => Promise<R | symbol>,
+    options: types.StrictOptions & types.TimeoutOptions & { omitAttached?: boolean },
+    action: (selector: SelectorInFrame | null, continuePolling: symbol) => Promise<R | symbol>,
     scope?: dom.ElementHandle): Promise<R> {
+    progress.log(`waiting for selector "${selector}"`);
     const continuePolling = Symbol('continuePolling');
     while (progress.isRunning()) {
-      const pair = await this._resolveFrameForSelector(progress, selector, options, scope);
-      if (!pair) {
-        // Missing content frame.
-        await new Promise(f => setTimeout(f, 100));
-        continue;
+      let selectorInFrame: SelectorInFrame | null;
+      if (options.omitAttached) {
```

Comment:
Somehow the name of this option is not clear to me. Perhaps it means "noWait" or "allowEmpty"?

Reply:
The semantics of this option did not change - well are as we were omitting the 'attached' requirement.